web/src/app/forms/FormContainer.js

```diff
@@ -117,50 +78,68 @@ export class FormContainer extends React.PureComponent {
         return i !== parseInt(idx, 10)
       })
 
-      return this.props.onChange(
+      return props.onChange(
         mapOnChangeValue(set(mapValue(oldValue), arrayPath, newArr)),
       )
     }
 
-    return this.props.onChange(
+    return props.onChange(
       mapOnChangeValue(set(mapValue(oldValue), fieldName, value)),
     )
   }
 
-  render() {
-    return <FormContext.Consumer>{this.renderComponent}</FormContext.Consumer>
-  }
+  return <FormContext.Consumer>{renderComponent}</FormContext.Consumer>
```

Comment:
Since we're in hooks now (good catch on your last comment) we can simplify a bit of this.

1. Instead of `renderComponent` being a function, just return it's value from the parent function
2. Add `const {disabled: formDisabled, addSubmitCheck} = useContext(FormContext)` to the top of the component function
3. Move the `useEffect` for `addSubmitCheck` to the top, right under the `useContext` call